pkg/executor/test/executor/executor_test.go

```diff
@@ -3088,3 +3088,66 @@ func TestQueryWithKill(t *testing.T) {
 	}()
 	wg.Wait()
 }
+
+func TestIssue58743(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("use test")
+	tk.MustExec(`
+		CREATE TABLE tlf5d55361 (
+			col_9 time NOT NULL,
+			col_10 float NOT NULL,
+			col_11 json NOT NULL,
+			col_12 date NOT NULL,
+			col_13 json NOT NULL,
+			col_14 tinyint NOT NULL,
+			col_15 date DEFAULT NULL,
+			col_16 tinyblob NOT NULL,
+			col_17 time DEFAULT '03:51:26',
+			PRIMARY KEY (col_14, col_9, col_10) /*T![clustered_index] CLUSTERED */
+		) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
+		PARTITION BY RANGE (col_14)
+		(
+			PARTITION p0 VALUES LESS THAN (-128),
+			PARTITION p1 VALUES LESS THAN (-84),
+			PARTITION p2 VALUES LESS THAN (-41),
+			PARTITION p3 VALUES LESS THAN (-30)
+		);
+	`)
+	tk.MustExec(`
+		CREATE TABLE td8d55878 (
+			col_26 datetime DEFAULT NULL,
+			col_27 time DEFAULT NULL,
+			col_28 json DEFAULT NULL,
+			col_29 char(186) CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT '4-BJKi',
+			col_30 date NOT NULL DEFAULT '1998-07-28',
+			col_31 datetime NOT NULL
+		) ENGINE=InnoDB DEFAULT CHARSET=gbk COLLATE=gbk_chinese_ci;
+	`)
+
+	tk.MustExec(`
```

Comment:
```suggestion
	tk.MustQuery(`
```

add result checking.